maxim/logger/smolagents/client.py

```
@@ -0,0 +1,539 @@
+    if isinstance(obj, (list, tuple, set)):
+        return [to_serializable_dict(item) for item in obj]
+    if hasattr(obj, 'dict') and callable(getattr(obj, 'dict')):
+        return to_serializable_dict(obj.dict())
+    if hasattr(obj, '__dict__'):
+        return to_serializable_dict(vars(obj))
+    if hasattr(obj, '_asdict') and callable(getattr(obj, '_asdict')):
+        return to_serializable_dict(obj._asdict())
+    return str(obj)
+
+
+def log_agent_execution(trace: Any, generation: Any, agent: Any, result: Any) -> None:
```

Ruff (0.12.2)

Unused function argument: `trace`

(ARG001)